### ISSUE TITLE
LocKey: Triage on the type ObjectDescriptor

### ISSUE THREAD

**Issue by SUMUKHA-PK** [SystemBuilders/LocKey#29]
* For argument: Since `LockDescriptor` wont have an `Owner` field once `sessions` have been implemented, there wont be any need for the `ObjectDescriptor` type.

* Against argument: As we have seen the need of the `ObjectDescriptor` type in the past, we might need to add more fields to `LockDescriptor` in the future and then we would need to revert back to having 2 types.

* Conflict: While calling `Acquire` or `Release` in the client, the argument - Descriptor, must not have an option for the user to send any sort of `OwnerID` on account of security. However, we need owner data further down the line because it needs to be passed on to the lockservice to verify that the user process that is manipulating the lock has access to do so.
Thus, the lockclient functions must have a different argument input - say `ObjectDescriptor` but `LockDescriptor` must stay so that it can be passed on later to the internal lockservice functions.

**Comment by suraj44** [comment]
@SUMUKHA-PK  I think LockDescriptor needs to keep the OwnerID field. Instead, the only argument to Acquire() will be the object. Internally, before sending the HTTP request to the server, the client can add the sessionID to the request.

**Comment by SUMUKHA-PK** [comment]
Yes that's what has happened now. Check the latest commit on the sessions PR. @suraj44
